File: java/src/Problem1752.java
public class Problem1752 {
    public boolean checkArray(int[] nums, int xRotate) {
        if (nums.length == 0) return true;
        int lastNumber = nums[(0 + xRotate)%nums.length];
        for (int arrayIdx = 1; arrayIdx < nums.length; arrayIdx++) {
            if (nums[(arrayIdx + xRotate)%nums.length] < lastNumber) return false;
            lastNumber = nums[(arrayIdx + xRotate)%nums.length];
        }
        System.out.println(xRotate);
        return true;
    }
    
    public boolean check(int[] nums) {
        for (int xRotate = 0; xRotate < nums.length; xRotate++) {
            if(checkArray(nums, xRotate)) {
                return true;
            }
        }
        return false;
    }

    public boolean check2(int[] nums) {
        if (nums.length <= 2) return true;
        int decreasingCount = 0;
        int lastNumber = nums[nums.length - 1];
        for (int numsIdx = 0; numsIdx < nums.length; numsIdx++) {
            if (nums[numsIdx] < lastNumber) decreasingCount++;
            if (decreasingCount > 1) return false;
            lastNumber = nums[numsIdx];
        }
        return true;
    }
}

File: c/problem-1752.c
#include <stdio.h>

int check(int* nums, int numsSize) {
    if (numsSize <= 2) return 1;
    int decreasingCount = 0;
    int lastNumber = nums[numsSize - 1];
    for (int numsIdx = 0; numsIdx < numsSize; numsIdx++) {
        if (nums[numsIdx] < lastNumber) decreasingCount++;
        if (decreasingCount > 1) return 0;
        lastNumber = nums[numsIdx];
    }
    return 1;
}

int main(void) {
    int nums[] = { 2,1,3,4 };
    int result = check(nums, 5);
    printf("Result: %d", result);
    return 0;
}